pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/schema/PulsarSchemaToKafkaSchema.java

```diff
@@ -80,6 +153,11 @@ private static org.apache.avro.Schema parseAvroSchema(String schemaJson) {
         return parser.parse(schemaJson);
     }
 
+    public static Schema getOptionalKafkaConnectSchema(org.apache.pulsar.client.api.Schema pulsarSchema) {
+        Schema s = getKafkaConnectSchema(pulsarSchema);
+        return new OptionalForcingSchema(s);
```

Comment:
we are creating a new object for each record.
maybe we could save this allocation by pushing down OptionalForcingSchema into the cache.
what do you think ?

Reply:
the only place it is used is within cache's get() call:

```
            return schemaCache.get(pulsarSchema.getSchemaInfo().getSchema(), () -> {
                if (pulsarSchema.getSchemaInfo().getType() == SchemaType.KEY_VALUE) {
                    KeyValueSchema kvSchema = (KeyValueSchema) pulsarSchema;
                    return SchemaBuilder.map(getKafkaConnectSchema(kvSchema.getKeySchema()),
                                             getKafkaConnectSchema(kvSchema.getValueSchema()))
                                    getOptionalKafkaConnectSchema(kvSchema.getValueSchema()))
                                .build();
                }
```